![logo_ironhack_blue 7](https://user-images.githubusercontent.com/23629340/40541063-a07a0a8a-601a-11e8-91b5-2f13e4e6b441.png)

# Lab | Hypothesis Testing


### Instructions

1. It is assumed that the mean systolic blood pressure is `μ = 120 mm Hg`. In the Honolulu Heart Study, a sample of `n = 100` people had an average systolic blood pressure of 130.1 mm Hg with a standard deviation of 21.21 mm Hg. Is the group significantly different (with respect to systolic blood pressure!) from the regular population?

   - Set up the hypothesis test.
   - Write down all the steps followed for setting up the test.
   - Calculate the test statistic by hand and also code it in Python. It should be 4.76190. What decision can you make based on this calculated value?

Hint: here is the statistic [table](https://yuppal.people.ysu.edu/econ_3790/t-table.pdf ) 

In [ ]:
# What is given:
# Population Mean: μ = 120 mm Hg
# Sample size: n = 100
# Sample mean: x° = 130.1 mm Hg
# Sample Standard Devaition: sd = 21.21 mm Hg

<b> Hiypothesis Test:

H0: Mean systolic blood pressure of the Honolulu is the same as the rest. μ = 120 mm Hg

H1: Mean systolic blood pressure of the Honolulu is significantly different than the rest. μ != 120 mm Hg

<b/> Steps to do the Hypothesis Test

Since the sample size is big enough, we can assume the normal distribution. 
- Step 1: Find out the critical Value. Since we have a stndard normal distribution and our confidence interval is 95%, Sigma 1 implies a critical value of |1,96|
- Step 2: Calculating the t-value. For that we need the Standrad deviation or error and insert the values in the formula
- Step 3: Comparing the critical & t-value 
- Step 4: Interpreting the results. Is the t-value bigger than the critical value, we reject the Null-hypothesis. If not, we failed to reject it. 

<b/> Computing the Hypothesis test

In [2]:
z = (130.1-120)/(21.21/10)
print("The z score of our sample is: {:.2f}".format(z))

The z score of our sample is: 4.76


In [4]:
import scipy.stats as stats
zc = stats.norm.ppf(0.95) # 0.95 is the confidence level.
print("The critical value corresponding to a 0.95 area of a normal distribution is: {:.2f}".format(zc))

The critical value corresponding to a 0.95 area of a normal distribution is: 1.64


Let's compare our z-score with the critical value:zc. Are we in the forbiden region?

z = 4.76 > zc = 1.64

-> We reject the null hypothesis. 
The people of Honolulu have significantly different bloodpressure than the rest

### Optional
2. In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file `Data/machine.txt`. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other.

Hint: use two sample t-test

t= (sample_mean(x1) - sample_mean(x2)) / sqrt(square(s1)/n1  +  square(s2)/n2 )